evalml/automl/automl_search.py

```diff
@@ -296,6 +306,14 @@ def __init__(self,
             parameters.update({'pipeline': self.problem_configuration})
             self._frozen_pipeline_parameters.update({'pipeline': self.problem_configuration})
 
+        self.sampler_method = sampler_method
+        self.sampler_balanced_ratio = sampler_balanced_ratio
+        self._sampler_name = None
+        if is_classification(self.problem_type):
+            self._sampler_name = self.sampler_method
+            if self.sampler_method == 'auto':
+                self._sampler_name = get_best_sampler_for_data(self.X_train, self.y_train, self.sampler_method, self.sampler_balanced_ratio)
```

Comment:
If `_sampler_name` is only used for making pipelines, maybe we should let `make_pipeline` handle the 'auto' to sampler name conversion? Or not store `_sampler_name` in the AutoMLSearch state?

Reply:
Fixing, but I will leave this in order to set the balanced ratio for the component